data_extractor/google_semantic_location_history/__init__.py

```diff
@@ -0,0 +1,106 @@
+"""Script to extract data from Google Semantic History Location zipfile"""
+__version__ = '0.1.0'
+
+import json
+import collections
+import itertools
+import re
+import zipfile
+
+import pandas as pd
+
+
+def __visit_duration(data):
+    """Get duration per visited place
+    Args:
+        data (dict): Google Semantic Location History data
+    Returns:
+        dict: duration per visited place, sorted in descending order
+    """
+    placevisit_duration = []
+    for data_unit in data["timelineObjects"]:
+        if "placeVisit" in data_unit.keys():
+            address = data_unit["placeVisit"]["location"]["placeId"]
+            start_time = data_unit["placeVisit"]["duration"]["startTimestampMs"]
+            end_time = data_unit["placeVisit"]["duration"]["endTimestampMs"]
+            placevisit_duration.append(
+                {address: (int(end_time) - int(start_time))/(1e3*24*60*60)})
+
+    # list of places visited
+    address_list = {list(duration.keys())[0] for duration in placevisit_duration}
```

Comment:
`next(iter(duration))` should do the same as `next(iter({"a": 1}))` without creating an intermediate list.